targettests/execution/cudaq_run_emulation.cpp

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// REQUIRES: remote-sim
+// REQUIRES: c++20
+
+// clang-format off
+// RUN: nvq++ -fenable-cudaq-run %cpp_std --target quantinuum --emulate  %s -o %t && %t | FileCheck %s
+// clang-format on
+
+#include <cudaq.h>
+
+__qpu__ int test_kernel(int count) {
+  unsigned result = 0;
+  cudaq::qvector v(count);
+  h(v[0]);
+  for (int i = 0; i < count - 1; i++)
+    cx(v[i], v[i + 1]);
+  for (int i = 0; i < count; i++)
+    if (mz(v[i]))
+      result += 1;
+  return result;
+}
+
+int main() {
+  int c = 0;
+  {
+    std::vector<int> results =
+        cudaq::run<int>(100, std::function<int(int)>{test_kernel}, 4);
```

Comment:
can you update this such that you don't require the template int parameter to be specified, e.g.
```cpp 
auto results = cudaq::run(100, test_kernel, 4); 
// -or-
std::vector<int> results = cudaq::run(100, test_kernel, 4); 
```
may just be that this is test code and we're being explicit, but I'd prefer the user not have to write this much boilerplate.

Reply:
Absolutely. It's on my radar. I'll tackle it in the following PR :)